onebusaway-android/src/main/java/org/onebusaway/android/ui/HomeActivity.java

```diff
@@ -2018,4 +2025,28 @@ private boolean isSlidingPanelCollapsed() {
     public ArrivalsListFragment getArrivalsListFragment() {
         return mArrivalsListFragment;
     }
+
+    private void setupLogUploadTask() {
+        PeriodicWorkRequest.Builder uploadLogsBuilder =
+                new PeriodicWorkRequest.Builder(NavigationUploadWorker.class, 24,
+                        TimeUnit.HOURS);
+
+        // Create the actual work object:
+        PeriodicWorkRequest uploadCheckWork = uploadLogsBuilder.build();
+
+        // Then enqueue the recurring task:
+        WorkManager.getInstance().enqueue(uploadCheckWork);
+    }
+
+    private void setupLogCleanupTask() {
+        PeriodicWorkRequest.Builder cleanupLogsBuilder =
```

Comment:
Let's try moving the setupLogCleanupTask() method after the trip finishes and after we set up the notification that the user can interact with.

Reply:
We have moved the `setupLogCleanupTask()` to `NavigationService.java` to enqueue the Work Manager after the trip finishes.